api.py

```diff
@@ -69,30 +69,68 @@
 #Sign_up
 @app.route("/auth/sign_up", methods=["GET", "POST"])
 def sign_up():
+    result =  auth.sign_up( request.form["firstname"], request.form["lastname"], request.form["email"], request.form["password"], request.form["confirmpassword"] )
```

Comment:
Actually, all this needs to be under the if statement `if request.method == "POST"`, because if it's a GET request, we don't want to create an account.

So I would put this line, as well as the `if "error" in result` statement under that if statement.

Also, make sure to put the `if "error" in result` statement **before** the `return result` line, or else it will never be run.